src/login/LoginOptionsScreen.tsx

```diff
@@ -0,0 +1,227 @@
+import FullScreenHeader from '@atb/components/screen-header/full-header';
+import {StyleSheet} from '@atb/theme';
+import {LoginTexts, useTranslation} from '@atb/translations';
+import React, {useEffect, useState} from 'react';
+import {ActivityIndicator, Linking, ScrollView, View} from 'react-native';
+import {useAuthState} from '@atb/auth';
+import ThemeText from '@atb/components/text';
+import {VippsSignInErrorCode} from '@atb/auth/AuthContext';
+import {RouteProp, useNavigation} from '@react-navigation/native';
+import {StaticColorByType} from '@atb/theme/colors';
+import {
+  getOrCreateVippsUserCustomToken,
+  authorizeUser,
+} from '@atb/api/vipps-login/api';
+import {useAppStateStatus} from '@atb/utils/use-app-state-status';
+import {parseUrl} from 'query-string';
+import storage from '@atb/storage';
+import {LoginInAppStackParams} from '@atb/login/in-app/LoginInAppStack';
+import {AfterLoginParams} from '@atb/login/types';
+import {VippsLoginButton} from '@atb/components/vipps-login-button';
+import MessageBox from '@atb/components/message-box';
+import * as Sections from '@atb/components/sections';
+
+const themeColor: StaticColorByType<'background'> = 'background_accent_0';
+
+export type LoginOptionsRouteParams = {
+  afterLogin: AfterLoginParams;
+};
+
+type LoginOptionsRouteProps = RouteProp<
+  LoginInAppStackParams,
+  'LoginOptionsScreen'
+>;
+
+type LoginOptionsProps = {
+  route: LoginOptionsRouteProps;
+};
+
+export default function LoginOptionsScreen({
+  route: {
+    params: {afterLogin},
+  },
+}: LoginOptionsProps) {
+  const {t} = useTranslation();
+  const styles = useThemeStyles();
+  const {signInWithCustomToken} = useAuthState();
+  const [error, setError] = useState<VippsSignInErrorCode>();
+  const [isLoading, setIsLoading] = useState(false);
+  const navigation = useNavigation();
+  const appStatus = useAppStateStatus();
+  const [authorizationCode, setAuthorizationCode] = useState('');
+
+  const authenticateUserByVipps = async () => {
+    setIsLoading(true);
+    try {
+      await authorizeUser(setIsLoading);
+    } catch (err) {
+      setError('unknown_error');
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(
+    () =>
+      navigation.addListener('focus', () => {
+        setIsLoading(false);
+      }),
+    [navigation],
+  );
+
+  const signInUsingCustomToken = async (token: string) => {
+    const response = await signInWithCustomToken(token);
```

Comment:
Hm. Due to the try/catch inside `signInWithCustomToken` and not returning the underlying promise, won't the error in line 76 always return`unknown_error` here? Meaning we'll never show specific error cases.

Reply:
Well yes, so this is the case where the user has given consent and we are in the after redirect to app flow. I am not sure if we will have any errors here that would make sense to the user. So we show a generic error to the user and ask them to try again.

Reply:
About the actual error, I think the response error handler in the axios client would notify bugsnag about it. Should we do more?

Reply:
I think there are cases where Vipps returns errors that the users should know about or that we should handle in some case: https://github.com/vippsas/vipps-login-api/blob/master/vipps-login-api.md#custom-error-codes

Reply:
Ah yeah. I read the flow wrong and thought of this as Vipps errors, but these are Firebase errors. There might be some errors there also that the user should know about, but also I was thinking more here https://github.com/AtB-AS/mittatb-app/pull/2848/files#diff-55e5a9f1f6d5f5e6abc246e3841ca0c8eb69c79e42d7b6e9746238d8f45d297fR91

Where it also seems maybe like it's just general "some error"?

Reply:
you are right, I did go through that list and thought of handling these errors but I don't remember why I ended up translating all of them to unknown_error.  🥺
```
export type VippsSignInErrorCode =
  | 'access_denied'
  | 'outdated_app_version'
  | 'unknown_error';
```
However, I only thought of handling these three errors. Also probably should handle `login_required`. But shouldn't rest of [them](https://github.com/vippsas/vipps-login-api/blob/master/vipps-login-api.md#custom-error-codes) translate to unknown_error to the end user? What do you think? Anyway, I think I should give this some more thought first and then get back 😅

Reply:
Yeah..I would atleast handle those then 👍

Reply:
I went through the code again and realised that these errors are already handled. They should be part of the query param in the callback URL if any of them occur. See https://github.com/AtB-AS/mittatb-app/pull/2848/files#diff-55e5a9f1f6d5f5e6abc246e3841ca0c8eb69c79e42d7b6e9746238d8f45d297fR114

As far as [this](https://github.com/AtB-AS/mittatb-app/pull/2848/files#diff-55e5a9f1f6d5f5e6abc246e3841ca0c8eb69c79e42d7b6e9746238d8f45d297fR91) is concerned, it should happen only when Bff is not able to redirect the user to the app due to some reason. And I think it should be nothing but an unknown error for the user. 

I think we should probably go with this error handling and see what scenarios we get during testing. cc @tormoseng  @HanneLH

Reply:
I see! Good catch.